### DISCUSSION TITLE
React 19 - Web Components

### DISCUSSION THREAD

**Discussion by Canic** [mantinedev/mantine#6595] [Q&A]
Hey,

We are using Mantine in our production environment and love the library.

With React 19 on the way and full support for Web Components, I have a question about the future of Mantine. Will there be support for Web Components in an upcoming version?

The background is that we are currently using React apps inside WordPress as widgets. Web Components would definitely make integration with pre-existing websites easier.

It would be great to hear what we can expect in the future.

**Comment by rtivital** [comment]
Mantine is a React components library, it is not planned to introduce web components

**Comment by Canic** [comment]
Hi,

there was a misunderstanding.
We don't want that mantine release standalone web components.

We have multiple standalone applications on a WordPress page using Mantine components. For example, one application is a contact form, while another might be just a search bar.

Our question is: with the release of React 19 and Web Components, does Mantine fully support it when we register a div as a web component? The background here is that it is not recommended to register the Mantine provider multiple times but would be then needed.

**Reply by rtivital** [reply]
There aren't any changes to Mantine that are related to React 19 yet. I also haven't seen any new web components features in React 19 upcoming release.

**Reply by Canic** [reply]
Hey,

I've created a small repository to showcase our usage of Mantine:
https://github.com/Canic/mantine-web-components-test

You'll see two web components defined, both utilizing Mantine.

Is there a recommended way to avoid initializing a new Mantine provider every time? Or perhaps a method to streamline it without complicating other parts of the website?

I really like Mantine and don't want to stop working with it.
